packages/zapp/console/src/components/Launch/LaunchForm/utils.ts

```diff
@@ -75,13 +75,25 @@ export function getTaskInputs(task: Task): Record<string, Variable> {
 /** Returns a formatted string based on an InputTypeDefinition.
  * ex. `string`, `string[]`, `map<string, number>`
  */
-export function formatType({ type, subtype }: InputTypeDefinition): string {
+export function formatType({ type, subtype, listOfSubTypes }: InputTypeDefinition): string {
   if (type === InputType.Collection) {
     return subtype ? `${formatType(subtype)}[]` : 'collection';
   }
   if (type === InputType.Map) {
     return subtype ? `map<string, ${formatType(subtype)}>` : 'map';
   }
+  if (type === InputType.Union) {
+    if (!listOfSubTypes) return 'union';
+
+    const concatListOfSubTypes = listOfSubTypes.reduce((previous, currentValue, currentIndex) => {
+      if (currentIndex === listOfSubTypes.length - 1) {
+        return previous + `${formatType(currentValue)}`;
+      }
+      return previous + `${formatType(currentValue)} |`;
+    }, '');
+
```

Comment:
curious if next would look a bit more clean (haven't tried locally):
```
const concatListOfSubTypes = listOfSubTypes.map(value => formatType(value)).join(' | ');
```
the speed difference should be minimal if any.

Reply:
i like your implementation it looks much cleaner

Reply:
updated